public/common/TracyTaggedUserlandAddress.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef __TRACYTAGGEDPTR_HPP__
+#define __TRACYTAGGEDPTR_HPP__
+
+#include <stdint.h>
+#include <assert.h>
+
+namespace tracy
+{
+
+class TaggedUserlandAddress
+{
+	uint64_t m_storage;
+	// So far no kernel seems to allow userspace address with more than 56bits userspace VAs since now hardware support it. See https://docs.kernel.org/next/x86/x86_64/mm.html and https://www.kernel.org/doc/html/v5.8/arm64/memory.html#bit-userspace-vas
+	static constexpr uint64_t kUserspaceVABits = 56;
+	static constexpr uint64_t kKernelOnlyBits = 64 - kUserspaceVABits;
+	static constexpr uint64_t kKernelOnlyBitMask = ( ( uint64_t(1) << kKernelOnlyBits ) - uint64_t(1) ) << kUserspaceVABits;
+public:
+	
+	TaggedUserlandAddress() = default;
+	explicit TaggedUserlandAddress( uint64_t address, uint8_t tag = 0 )
+	{
+		SetAddress( address );
+		SetTag( tag );
```

Comment:
When constructing, we should just calculate the value of `m_storage` directly, using given inputs. Both these functions want to preserve the other part of the variable, which in case of the first call is garbage, and for the second call can be computed more easy.